frontend/src/components/Dashboard/ServiceTreeView.vue

```diff
@@ -0,0 +1,250 @@
+                  @click.prevent="handleFilterSearch"
+                >
+                  {{ $t('search') }}
+                </button>
+                <button
+                  type="button"
+                  class="btn btn-outline-secondary"
+                  @click.prevent="handleClearFilter"
+                >
+                  Reset
+                </button>
+              </div>
+            </div>
+          </div>
+        </form>
+      </div>
+
+      <div class="col-sm-6">
+        <ul class="d-flex justify-content-end align-items-center">
+          <li class="d-flex">
+            <div class="mr-1 text-shade-success">
+              <FontAwesomeIcon
+                icon="circle"
+                class="border border-secondary rounded-circle"
+              />
+            </div>
+            <div>Up</div>
+          </li>
+          <li class="d-flex ml-3">
+            <div class="mr-1 text-shade-warning">
+              <FontAwesomeIcon
+                icon="circle"
+                class="border border-secondary rounded-circle"
+              />
+            </div>
+            <div>Degraded</div>
+          </li>
+          <li class="d-flex ml-3">
+            <div class="mr-1 text-shade-danger">
+              <FontAwesomeIcon
+                icon="circle"
+                class="border border-secondary rounded-circle"
+              />
+            </div>
+            <div>Down</div>
+          </li>
+        </ul>
+      </div>
+    </div>
+
+    <div class="mt-3">
+      <div v-if="isLoading">
+        <div class="col-12 text-center">
+          <FontAwesomeIcon
+            icon="circle-notch"
+            size="3x"
+            spin
+          />
+        </div>
+        <div class="col-12 text-center mt-3 mb-3">
+          <span class="text-muted">
+            Loading Services
+          </span>
+        </div>
+      </div>
+    
+      <ul
+        v-else
+        class="parent-list-group pl-0 mb-0 overflow-auto"
+      >
+        <TreeItem
+          v-for="service in treeData"
+          :key="service.id"
+          :item="service"
+        />
+      </ul>
+    </div>
+  </div>
+</template>
+
+<script>
+import TreeItem from '../Elements/TreeItem.vue';
+import FlatPickr from 'vue-flatpickr-component';
+import 'flatpickr/dist/flatpickr.css';
+
+const getRootNodes = (data) => {
+    if (!data || data.lenght === 0) {
+        return;
+    }
+
+    const rootNode = data.reduce((acc, service) => {
+        const isChild = data.find((item) => {
+            if (item.sub_services_detail) {
+                return Object.keys(item.sub_services_detail).includes(String(service.id));
+            }
+
+            return false;
+        });
+
+        if (!isChild) {
+            acc.push(service);
+        }
+
+        return acc;
+    }, []);
+
+    return rootNode;
+};
+
+const getTreeData = (parentServices, serviceStatus) => {
+    const treeData = [];
+
+    for (let i=0; i<parentServices.length; i++) {
```

Comment:
**nitpick (readability):** - any particular reason of favoring for loops instead of `parentServices.forEach(parentService => {})`

Feel free to to take this as suggestion. If not, then `parentServices[i]` is being referenced multiple times – consider assigning it to a variable at the start of loop.